### PR TITLE
[Wallet] Allow users to buy CELO on mainnet through Moonpay

### LINKED ISSUE #4214: [Wallet] Users SBAT use production Moonpay Cash In
### Expected Behavior

Production environment is used by updating API keys in the /moonpay-auth cloud function and repointing wallet to that function. If cUSD is not yet live on Moonpay, will also require switch to CELO in Cash In flow

### Current Behavior

Staging environment is used with cUSD

### PR DESCRIPTION
**Pull request by annakaz**
### Description

Update Moonpay widget URL to production, so that users can buy CELO in the Cash In flow. 

### Other changes

Added message explaining that Celo dollars are not yet available, but can be obtained by exchanging for CELO, based on a CUSD_MOONPAY_ENABLED feature flag:

Screen Shot 2020-07-21 at 4 29 04 PM

This PR keeps the Cash In input as the local currency displayed in dollars- still in discussion with @nityas on a possible change here

Added signMoonpayStaging to maintain the old staging API keys and setup that we will want to use to test the cUSD integration. 

### Tested

Confirmed that Moonpay widget loaded with expected values pre-filled with the new `signMoonpayProd` function deployed to `celo-org-mainnet`. Confirmed that API keys from the encrypted .env file were properly loaded to firebase. 

Cannot test full Moonpay flow as CELO is not available in the US via Moonpay (waiting to hear back from them on what is holding this up). 

### Related issues

- Fixes #4214 

### Backwards compatibility

Yes, the previous signMoonpay function remains deployed.

### CODE REVIEW

**Review by i1skn — APPROVED** [review]
🚢 it!

File: packages/moonpay-auth/package.json [hunk 1]
```diff
@@ -7,9 +7,10 @@
   "main": "dist/index.js",
   "scripts": {
     "serve": "firebase serve --only functions",
-    "copy-env-to-firebase": "firebase functions:config:set envs.public_key=$(grep PUBLIC_KEY .env | cut -d '=' -f 2-) envs.private_key=$(grep PRIVATE_KEY .env | cut -d '=' -f 2-)",
-    "predeploy": "yarn copy-env-to-firebase",
-    "deploy": "firebase deploy --only functions",
+    "update-firebase-config:prod": "firebase functions:config:set envs.public_key_staging=$(grep PUBLIC_KEY_STAGING .env | cut -d '=' -f 2-) envs.secret_key_staging=$(grep SECRET_KEY_STAGING .env | cut -d '=' -f 2-)",
+    "update-firebase-config:staging": "firebase functions:config:set envs.public_key_prod=$(grep PUBLIC_KEY_PROD .env | cut -d '=' -f 2-) envs.secret_key_prod=$(grep SECRET_KEY_PROD .env | cut -d '=' -f 2-)",
+    "deploy:prod": "firebase use prod && yarn update-firebase-config:prod && firebase deploy --only functions:signMoonpayProd",
+    "deploy:staging": "firebase use default && yarn update-firebase-config:staging && firebase deploy --only functions:signMoonpayStaging",
```
Comment (jeanregisser) [inline packages/moonpay-auth/package.json:13]: no change needed now, but noting we can probably start using a similar pattern as what we have for `blockchain-api` / `notification-service` with a deploy script taking the network name as parameter.

File: packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ import BackButton from 'src/components/BackButton.v2'
 import CurrencyDisplay from 'src/components/CurrencyDisplay'
 import LineItemRow from 'src/components/LineItemRow'
 import { DOLLAR_ADD_FUNDS_MIN_AMOUNT, DOLLAR_CASH_OUT_MIN_AMOUNT } from 'src/config'
+import { CUSD_MOONPAY_ENABLED } from 'src/flags'
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx]: I think this doesn't work, we have to `import { features } from 'src/flags` and then use `features.CUSD_MOONPAY_ENABLED`.

File: packages/moonpay-auth/src/index.ts [hunk 3]
```diff
@@ -1,25 +1,71 @@
...
+  MOONPAY_SECRET_KEY_PROD,
+  MOONPAY_SECRET_KEY_STAGING,
+  MOONPAY_URL_PROD,
+  MOONPAY_URL_STAGING,
+} from './config'
 const URL = require('url').URL
 
-export const signMoonpay = functions.https.onRequest((request, response) => {
+// TODO: This staging function is left in place to test the cUSD integration
+// that we are waitin on from Moonpay. It will be removed after
```
Comment (jeanregisser) [inline packages/moonpay-auth/src/index.ts]: nit: typo, waitin**g**

File: packages/mobile/src/geth/networkConfig.ts [hunk 4]
```diff
@@ -28,6 +28,8 @@ const pgpnpPubKeyStaging =
   '7FsWGsFnmVvRfMDpzz95Np76wf/1sPaK0Og9yiB+P8QbjiC8FV67NBans9hzZEkBaQMhiapzgMR6CkZIZPvgwQboAxl65JWRZecGe5V3XO4sdKeNemdAZ2TzQuWkuZoA'
 
 const signMoonpayUrlStaging = 'https://us-central1-celo-org-mobile.cloudfunctions.net/signMoonpay'
```
Comment (jeanregisser) [inline packages/mobile/src/geth/networkConfig.ts]: function name for staging should be changed to `signMoonpayStaging` too.

**Review by jeanregisser — CHANGES_REQUESTED** [review]
Looking almost good!! 🎉

**Review by jeanregisser — APPROVED** [review]
I went ahead and committed small tweaks so we can get it merged sooner 👍

**Comment by jeanregisser** [comment]
I also deployed `https://us-central1-celo-org-mobile.cloudfunctions.net/signMoonpayStaging`

File: packages/mobile/src/geth/networkConfig.ts (hunk 4 above)
Comment (annakaz) [reply packages/mobile/src/geth/networkConfig.ts]: Thanks @jeanregisser ! I had left `signMoonpay` deployed to be compatible with existing clients (although it isn't being used), but thinking about this more it's important to have whatever is deployed match the code.

**Comment by annakaz** [comment]
Thanks for getting this merged 😄

**Pull request merged**